Etg.Yams/Storage/IYamsRepository.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Threading.Tasks;
+using Etg.Yams.Application;
+using Etg.Yams.Storage.Config;
+
+namespace Etg.Yams.Storage
+{
+    public interface IYamsRepository
```

Comment:
Few concerns about `IYamsApplicationRepository`:
* A bit long 
* Application is a bit redundant because it appears in the method names (e.g. UploadApplicationBinaries). 
* Not clear whether it's the Repository for one Application or many (Applications).  
* The repository contains more than application binaries (DeploymentConfig.json and maybe more in the future).

Thoughts? @ReubenBond @galvesribeiro ?

Other possible names I can think of, IApplicationRepository, IApplicationsRepository, IYamsStorage, IDeploymentRepository, IDeploymentsRepository.

I like the current name though :)